smithy-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ClientConfigGeneratorTest.kt

```diff
@@ -77,6 +88,11 @@ public class Config private constructor(builder: Builder): HttpClientConfig, Ide
          * debug purposes.
          */
         public var sdkLogMode: SdkLogMode = SdkLogMode.Default
+        /**
+         * The probe that receives tracing events such as logging messages and metrics. This probe can be used
+         * to send tracing events to other frameworks outside the SDK. By default, a no-op probe is selected.
+         */
+        public var traceProbe: TraceProbe? = null
```

Comment:
should we be taking a probe here or should we be taking a `TraceSpan` and defaulting it to a `DefaultTracer` with `NoOpTraceProbe`? 

If we leave it as is (which I think is a reasonable path) it means we are always the root span which means the SDK clients can belong to some other parent span. This is probably ok since we don't really expect customers to instrument their code with our tracing abstractions but worth some thought.